### ISSUE TITLE
Only the last single-line comment within an array is emitted

### ISSUE THREAD

**Issue by jmlee2k** [microsoft/TypeScript#45776]
# Bug Report

### 🔎 Search Terms

comments, preserve, emit, array

### 🕗 Version & Regression Information

I noticed it in 4.4.2, but it seems to have happened in some form since 3.3.2 (oldest available on the playground).
The behavior appeared to change slightly after 4.2.3. (See playground link for details).

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about emitting comments

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.4.2#code/PTD2DsBsE8AIBcAWBTWBnZBjCATW2BbA5ceWAdwEtJJYAjVZAy+eZHAKG3DTIEMATgICMsALywA2h1izYwgDSwQAM0oDeMuQCZlwDN04BdDh2AAqWJD69YLJrEQ3YfBAL7VK4AOb5QRPi1ZABUAZVgAHlgAFgA6bViAZlgAWlhwUD8iEng0CmpaBlgmFjZOOVgw2AA+CTiE5LSIGAQUdCxcLOJSfJp6RmZWdlhzYFguCFtBEQB1FkRg908fAGF-AlcJaQrFPTUNeCDYbSUQA1wOEzMwKDgkVHPwPEJusio+opKhzm4poV0tkddhZ9rxRkddBZHjhwVcQHRQEgujk8u9CgNSuwJjx+EJklthCD1GDgAptFCOk9RkYgA)

### 💻 Code

```ts
const arr1 = [
    1, //first
    2 //second
]
```
emitted as
```js
"use strict";
const arr1 = [
    1,
    2 //second
];
```

### 🙁 Actual behavior

Only the last (second) comment is emitted

### 🙂 Expected behavior

I would expect all comments to appear in the output JS

**Comment by YaseenYo** [comment]
Hello @jmlee2k, I want to contribute with this issue. I am new to open source. Please can you help me with this..

**Comment by AndyClausen** [comment]
I found some more cases where comments are not emitted. It seems like it's the exact same behavior for objects, but function parameters have the complete opposite issues if you have a trailing comma, i.e. everything _but_ the last comment is emitted.

Here is the [updated playground](https://www.typescriptlang.org/play?ts=3.3.3#code/PTD2DsBsE8AIBcAWBTWkCGBnesDGoBbA5cHAdwEtJJYAjVZAi+eZAEwCh9xtZ0AnfgEZYAXlgBtDrBmwhAGlggAZhX7ZpsgEyKQmZN06zYAZiXAkazgF0OHYACo0WHM0axEWPgn7oqFcABzPEICdE0ZABUAZVgAHlgAFgA6LWSzAFpYcFAQohJ4TFhKajoGJhZ2CNgY2AA+cRS0zNgIGAQUZ158fNJiqhp6WEZmVjZYB2BYLgheAWEAdWZESN9-IIBhUPQxSWqFc1V1eGqdc31DapNdC0QrDlt7MCg4JFRLfnGe4j6SwfLRlVuHNBFpdlJjAdHEdsJNTroHBcIGw4cYzI4keAUcAHnYQLRQEg8j9Cv1SkMRpVOMCcPMzOIJEJoWpYcB5FoMQZkZNHvjCYhiQUitgBsMKmN+kTvjt6MpQPx3p1vgUZjxaYJEuCFMzjpNYBzEVysTy8cBMOhiLA5fxWrQAFYGQqq3ige0icQAb2q6AAXLAAET+m4wk7GWh+wM3TFGWS4CP+8wfTgAX2dOFddrBnu98eDLNDsnDAaD5yNMZkceLNyTHFTT2UAFdwLh4BQILAAA4CC3IVjqMXMIroUrK0iYDiN5ut9vKcAAChzfHA0Dzx2qRfQy6jZeqlc3cBANeMAEpYF7jAr4A3+OAANy101NyDITBFaCgBsedAAN1QO3gayQAEwTSq0LwDqSIZLuM0YTk2LZtuAVrgEsSCrH4QGbNsC7GL6S4roc+brn6+7bpcxh7luiZ3J81SnuesiXted4PkAA).

**Comment by HuijungYoon** [comment]
Hello @jmlee2k @andrewbranch. i am huijung. 
if it has't been solved. i want to contribute with this issue.  thanks a lot

**Comment by danvk** [comment]
This looks like it was fixed between 5.2.2 and 5.3.3.

- [5.2.2 Playground][5.2.2] with some comments missing from emitted JS
- [5.3.3 Playground][5.3.3] with all comments emitted

[5.2.2]: https://www.typescriptlang.org/play?ts=5.2.2#code/PTD2DsBsE8AIBcAWBTWBnZBjCATW2BbA5ceWAdwEtJJYAjVZAy+eZHAKG3DTIEMATgICMsALywA2h1izYwgDSwQAM0oDeMuQCZlwDN04BdDh2AAqWJD69YLJrEQ3YfBAL7VK4AOb5QRPi1ZABUAZVgAHlgAFgA6bViAZlgAWlhwUD8iEng0CmpaBlgmFjZOOVgw2AA+CTiE5LSIGAQUdCxcLOJSfJp6RmZWdlhzYFguCFtBEQB1FkRg908fAGF-AlcJaQrFPTUNeCDYbSUQA1wOEzMwKDgkVHPwPEJusio+opKhzm4poV0tkddhZ9rxRkddBZHjhwVcQHRQEgujk8u9CgNSuwJjx+EJklthCD1GDgAptFCOk9RkYgA
[5.3.3]: https://www.typescriptlang.org/play?ts=5.3.3#code/PTD2DsBsE8AIBcAWBTWBnZBjCATW2BbA5ceWAdwEtJJYAjVZAy+eZHAKG3DTIEMATgICMsALywA2h1izYwgDSwQAM0oDeMuQCZlwDN04BdDh2AAqWJD69YLJrEQ3YfBAL7VK4AOb5QRPi1ZABUAZVgAHlgAFgA6bViAZlgAWlhwUD8iEng0CmpaBlgmFjZOOVgw2AA+CTiE5LSIGAQUdCxcLOJSfJp6RmZWdlhzYFguCFtBEQB1FkRg908fAGF-AlcJaQrFPTUNeCDYbSUQA1wOEzMwKDgkVHPwPEJusio+opKhzm4poV0tkddhZ9rxRkddBZHjhwVcQHRQEgujk8u9CgNSuwJjx+EJklthCD1GDgAptFCOk9RkYgA
